sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -856,7 +857,90 @@ trait GpuAggregateWindowFunction[T <: Aggregation with RollingAggregation[T]]
   def windowAggregation(inputs: Seq[(ColumnVector, Int)]): AggregationOnColumn[T]
 }
 
-case class GpuRowNumber() extends GpuAggregateWindowFunction[RowNumberAggregation] {
+trait BatchedRunningWindowFixer extends AutoCloseable {
+  /**
+   * Save any state needed from the previous output that will be needed to fix up the next batch.
+   * You don't need to worry about the partition columns. Those will be saved separately.
+   * @param finalOutputColumn the output of fixup.
+   */
+  def updateState(finalOutputColumn: GpuColumnVector): Unit
+
+  /**
+   * Fix up `windowedColumnOutput` with any stored state from previous batches.
+   * @param samePartitionMask a boolean mask with no nulls in it. `true` indicates the row
+   *                          is for the same partition by keys that was the last row in the
+   *                          previous batch. `false` indicates it is not. Only values that are for
+   *                          the same partition by keys should be modified.
+   * @param windowedColumnOutput the output of the windowAggregation without anything
+   *                             fixed/modified.
+   * @return a fixed ColumnVector that was with outputs updated for items that were in the same
+   *         group by key as the last row in the previous batch.
+   */
+  def fixUp(
+      samePartitionMask: Either[GpuColumnVector, Boolean],
+      windowedColumnOutput: GpuColumnVector): GpuColumnVector
+}
+
+/**
+ * For many operations a running window (unbounded preceding to current row) can
+ * process the data without dividing the data up into batches that contain all of the data
+ * for a given group by key set. Instead we store a small amount of state from a previous result
+ * and use it to fix the final result. This is a memory optimization. Any
+ * GpuAggregateWindowFunction can still work for running window queries. This just reduces the
+ * maximum amount of memory needed to process them.
+ */
+trait GpuBatchedRunningWindowFunction[T <: Aggregation with RollingAggregation[T]]
+    extends GpuAggregateWindowFunction[T] {
+
+  /**
+   * Get a new class that can be used to fix up batched RunningWindowOperations.
+   */
+  def newFixer(): BatchedRunningWindowFixer
```

Comment:
Seems like a "fixer" is an aggregate function used to combine two partial values from the same key. Could there be a more descriptive name than "fixer"? Wondering if something like "combiner" captures its function, but I am not entirely sure yet.

Reply:
Aggregate function to me means it is doing an aggregation. This is not doing an aggregation. It is fixing up the result of an already done aggregation based off of data that aggregation was missing. Spark already has an `AggregateFunction` so I really want to avoid that term or anything like it. `Combiner` shows up in Spark as a part of shuffle (Which is the same as in Hadoop it lets you do an aggregation as you shuffle the data), and as a more generic aggregation API for RDDs.  I guess this could be called a `Combiner`, but it does not really fit. A combiner typically is a function that takes two of the same type of value and outputs a new one. It is intended to be used in an iterative reduction/aggregation.  This is not that.  Which is why I called it a "Fixer". I appreciate naming help and I agree "Fixer" is not a good name, but that is why it is a `BatchedRunningWindowFixer` to try and make it less ambiguous.

Reply:
Agree combine is not a good a choice either. Regardless of name, it doesn't seem entirely clear why we need to fix batches up. Could a short example be included in a comment on how the fixing process works? This is the reason why I was trying to suggest a different name, I am not entirely sure why we are fixing and what the fix is.